mne_lsl/utils/_tests.py

```diff
@@ -29,14 +29,20 @@ def sha256sum(fname):
 
 def match_stream_and_raw_data(data: NDArray[float], raw: BaseRaw) -> None:
     """Check if the data array is part of the provided raw."""
-    for start in range(raw.times.size):
-        if np.allclose(np.squeeze(raw[:, start][0]), data[:, 0], atol=0, rtol=1e-8):
-            break
-    else:
-        raise RuntimeError("Could not find match between data and raw.")
+    # Sample numbers should be in the last row of the data, but could loop so are
+    # not necessarily always increasing by 1
+    good = np.isclose(raw[:][0], data[:, :1], atol=1e-10, rtol=1e-8).all(axis=0)
```

Comment:
I thought I had refactored this to actually find the sample number from the `data` stream's last channel. It should make it easy to detect stuff like skipped buffers in `data`. I'll look to see if my memory is correct, but if you did indeed change this back to an `allclose`-based check let me know what the rationale for using that was. It's inexact and less clean than just forcing `raw` and `data` to both contain sample numbers as a data channel I think, even though that itself is a bit of a hack...

Reply:
Ahh nevermind this is just as I left it... the sample numbers *should* still be in the last row of data. I'll see if `data` is missing samples for example, it would explain at least one macOS failure.

Is it posssible it's a dropped "packet" at the LSL end somehow? We might need to add more `logger.debug` statements when the virtual streaming layer sends packets or whatever

Reply:
Yes, I liked that hack, it was elegant and better than the `allclose`-based check.
I don't think it's a dropped packet at the LSL-end, it should resend incomplete package, but maybe.. I'll try to ask on the LSL Slack tomorrow.

Reply:
This is really weird. The sample numbers match... but the data doesn't, at least according to my new check (it checks the sample numbers for equivalence before the data itself):

https://github.com/mne-tools/mne-lsl/actions/runs/6855412358/job/18640383219#step:9:1873

Maybe some weird floating point or inplace operation thing happening?

Is data multiplied inplace anywhere in the code, e.g., by a scale factor, then multplied back? Maybe a `flags["WRITEABLE"] = False` some places would help

Reply:
... in case somehow the data is being multiplied and divided in one thread, and accessed by another one simultaneously

Reply:
Found one more issue, the sample number in the last channel does not work because as soon as you add a reference channel, it's added last. I'm moving it to the first channel instead. I'll have a look for the floating point/in-place operation. I'll push those changes today :crossed_fingers: